simson/steel/steel_trade_model.py

```diff
@@ -0,0 +1,96 @@
+import sys
+
+from sodym.trade import ScalingBalancedTrade, MinMaxBalancedTrade
+from sodym import Parameter
+from simson.common.data_transformations import extrapolate_to_future
+from pydantic import BaseModel as PydanticBaseModel
+
+class SteelTradeModel():
+    """A trade model for the steel sector storing the data and defining how trade is processed."""
+
+    def __init__(self, dims, trade_data):
+        self.historic_trade = {}
+        self.future_trade = {}
+        self.dims = dims
+
+        self.historic_trade['Intermediate'] = MinMaxBalancedTrade(dims=dims,
+                                                imports=trade_data['direct_imports'],
+                                                exports=trade_data['direct_exports'])
+        self.historic_trade['Indirect'] = MinMaxBalancedTrade(dims=dims,
+                                            imports=trade_data['indirect_imports'],
+                                            exports=trade_data['indirect_exports'])
+        self.historic_trade['Scrap'] = MinMaxBalancedTrade(dims=dims,
+                                            imports=trade_data['scrap_imports'],
+                                            exports=trade_data['scrap_exports'])
+
+    def balance_historic_trade(self):
+        for trade in self.historic_trade.values():
+            trade.balance(by='maximum')
+
+    def balance_future_trade(self):
+        for name, trade in self.future_trade.items():
+            trade.balance()
+
+    def predict(self, future_in_use_stock):
+        for trade in self.historic_trade.values():
+            assert 'h' in trade.imports.dims.letters and 'h' in trade.exports.dims.letters, \
+                "Trade data must have a historic time dimension."
+
+        # establish future trade as a ScalingBalancedTrade
+
+        for name, trade in self.historic_trade.items():
+            new_dims = trade.imports.dims.replace('h', self.dims['t'])
+            if 'Scrap' == name:
+                new_dims = new_dims.expand_by([self.dims['g']])
+            self.future_trade[name] =  ScalingBalancedTrade(dims=self.dims,
+                                                            imports=Parameter(name=name, dims=new_dims),
+                                                            exports=Parameter(name=name, dims=new_dims))
+
+        # predict different kinds of trade
+
+        product_demand = future_in_use_stock.inflow
+        eol_products = future_in_use_stock.outflow
+
+        self.predict_indirect_trade(product_demand)
+        self.predict_intermediate_trade(product_demand)
+        self.predict_scrap_trade(eol_products)
```

Comment:
I think it would also be nice to have the predict method similar to the balance method above, i.e. just loop through all the three Trade's and run predict on each of them. Then, we could think about having predict as a method of the Trade class. I'm actually wondering now, if the Trade class would be more flexible like this:
```
class Trade(pydantic.BaseModel):
    imports: ...
    exports: ...
    balancer: Optional[callable]  # Could be `callable`  = a function, or we define a class for this
    predictor: Optional[callable]  # Could be `callable`  = a function, or we define a class for this

    def balance():
        # balance imports and exports using the balancer object/function (if available, else raise error)
        # if inplace update values, else return new object 

    def predict(self, future_time):
        # predict future imports and exports using the predictor object/function (if available, else raise error)
        # return a new class object
```
I know it was my suggestion to have the balancer function defined within the Trade class, but I have changed my mind a bit, sorry. I think this way would be more flexible. Let me know what you think. Happy to have a call to discuss!